relint.py

```diff
@@ -46,17 +52,73 @@ def load_config(path):
             )
 
 
-def lint_file(filename, tests):
-    try:
-        with open(filename) as fs:
-            content = fs.read()
-    except (IsADirectoryError, UnicodeDecodeError):
-        pass
-    else:
+def lint_file(filename, tests, diff=None):
+    if diff and diff.get(filename):
         for test in tests:
-            if any(fnmatch.fnmatch(filename, fp) for fp in test.filename):
+            for line_number, content in diff.get(filename).items():
                 for match in test.pattern.finditer(content):
-                    yield filename, test, match
+                    yield filename, test, match, line_number
+    else:
+        try:
+            with open(filename) as fs:
+                content = fs.read()
+        except (IsADirectoryError, UnicodeDecodeError):
+            pass
+        else:
+            for test in tests:
+                if any(fnmatch.fnmatch(filename, fp) for fp in test.filename):
+                    for match in test.pattern.finditer(content):
+                        yield filename, test, match, None
+
+
+def parse_diff(output):
+    changed_content = {}
+    current_file = None
+    line_after_number_line = False
+    current_line = None
+
+    pattern = re.compile(r"(([+]|[-])\d*[,]?\d*)")
+
+    for line in output.splitlines():
+        if line.startswith('diff'):
+            current_file = line[line.rfind(' b/')+3:]
+            line_after_number_line = False
+
+        elif line.startswith('@@'):
+            result = re.findall(pattern, line)
```

Comment:
```suggestion
            result = pattern.findall(line)
```
see also https://docs.python.org/3/library/re.html#re.Pattern.findall